### PR TITLE
Fix for issue #3596.

### PR DESCRIPTION
**Pull request by Azureblade3808**
I tried to make a fix for issue #3596 and add a test case directed to it. Please check.

### CODE REVIEW

File: ReactiveCocoa/NSObject+KeyValueObserving.swift [hunk 1]
```diff
@@ -223,6 +223,11 @@ extension KeyValueObserver {
 		if isNested {
 			observer = KeyValueObserver(observing: object, key: keyPathHead, options: options.union(.initial)) { object in
 				guard let value = object?.value(forKey: keyPathHead) as! NSObject? else {
+					if let headDisposable = headSerialDisposable.inner {
+						headDisposable.dispose()
+						headSerialDisposable.inner = nil
```
Comment (andersio) [inline ReactiveCocoa/NSObject+KeyValueObserving.swift]: Setting it to `nil` is sufficient given the semantic of `SerialDisposable`.

**Comment by Azureblade3808** [comment]
I haven't used `SerialDisposable` in practice and am not familiar with it. Thanks for letting me know the proper usage.

**Comment by Azureblade3808** [comment]
Changes have been done.

**Comment by mdiep** [comment]
Awesome! Thanks for fixing this @Azureblade3808!

**Pull request merged**